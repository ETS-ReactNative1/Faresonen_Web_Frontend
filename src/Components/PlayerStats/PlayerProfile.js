import React, { Component } from "react";
import PropTypes from "prop-types";
import { Card, Image, Icon } from "semantic-ui-react";

const PlayerProfile = ({
  name,
  news,
  photo,
  chanceOfPlayingNextRound,
  inDreamTeam,
  cost,
  selectedBy,
  points
}) => {
  return (
    <div>
      <Card>
        <Image
          src={
            photo
              ? `http://platform-static-files.s3.amazonaws.com/premierleague/photos/players/250x250/p${photo.replace(
                  "jpg",
                  "png"
                )}`
              : ""
          }
        />
        <Card.Content>
          <Card.Header>
            {name} {inDreamTeam ? <Icon name="star" color="yellow" /> : ""}
          </Card.Header>
          <div>
            <Card.Description>
              <b>Pris: </b>£{cost / 10}
            </Card.Description>
            <Card.Description>
              <b>Poeng: </b>
              {points}
            </Card.Description>
            <Card.Description>
              <b>Valgt av: </b>
              {selectedBy}%
            </Card.Description>
            <Card.Description>
              <b>Skader: </b>
              {news ? `${news}` : "Ingen skader."}
            </Card.Description>
            {chanceOfPlayingNextRound ? (
              <Card.Description>
                <b>Sjanse for å spille neste runde:</b>{" "}
                {chanceOfPlayingNextRound}%
              </Card.Description>
            ) : (
              ""
            )}
          </div>
        </Card.Content>
      </Card>
    </div>
  );
};
export default PlayerProfile;
