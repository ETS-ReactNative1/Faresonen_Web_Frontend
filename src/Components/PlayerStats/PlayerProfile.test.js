import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import PlayerProfile from "./PlayerProfile";

describe("<PlayerProfile />", () => {
  it("Should render properly", () => {
    let tree = renderer.create(
      <PlayerProfile
        name="Marcos Alonso"
        teamName="Chelsea"
        news=""
        photo="p82263.jpg"
        chanceOfPlayingNextRound={100}
        inDreamTeam
        cost={72}
        selectedBy="19,7"
        points={146}
      />
    );
    expect(tree).toMatchSnapshot();
    tree = renderer.create(
      <PlayerProfile
        name="Marcos Alonso"
        teamName="Chelsea"
        news=""
        cost={72}
        selectedBy="19,7"
        points={146}
      />
    );
    expect(tree).toMatchSnapshot();
    tree = renderer.create(<PlayerProfile name="Marcos Alonso" />);
    expect(tree).toMatchSnapshot();
  });
});
