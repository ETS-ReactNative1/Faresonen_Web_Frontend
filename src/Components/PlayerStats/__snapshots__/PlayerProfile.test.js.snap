// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerProfile /> Should render properly 1`] = `
<div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <img
      className="ui image"
      src="http://platform-static-files.s3.amazonaws.com/premierleague/photos/players/250x250/pp82263.png"
    />
    <div
      className="content"
    >
      <div
        className="header"
      >
        Marcos Alonso
        ,
        Chelsea
         
        <i
          aria-hidden="true"
          className="yellow star icon"
          onClick={[Function]}
        />
      </div>
      <div>
        <div
          className="description"
        >
          <b>
            Pris: 
          </b>
          £
          7.2
        </div>
        <div
          className="description"
        >
          <b>
            Poeng: 
          </b>
          146
        </div>
        <div
          className="description"
        >
          <b>
            Valgt av: 
          </b>
          19,7
          %
        </div>
        <div
          className="description"
        >
          <b>
            Skader: 
          </b>
          Ingen skader.
        </div>
        <div
          className="description"
        >
          <b>
            Sjanse for å spille neste runde:
          </b>
           
          100
          %
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PlayerProfile /> Should render properly 2`] = `
<div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <img
      className="ui image"
      src=""
    />
    <div
      className="content"
    >
      <div
        className="header"
      >
        Marcos Alonso
        ,
        Chelsea
         
        
      </div>
      <div>
        <div
          className="description"
        >
          <b>
            Pris: 
          </b>
          £
          7.2
        </div>
        <div
          className="description"
        >
          <b>
            Poeng: 
          </b>
          146
        </div>
        <div
          className="description"
        >
          <b>
            Valgt av: 
          </b>
          19,7
          %
        </div>
        <div
          className="description"
        >
          <b>
            Skader: 
          </b>
          Ingen skader.
        </div>
        <div
          className="description"
        >
          <b>
            Sjanse for å spille neste runde:
          </b>
           
          -1
          %
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PlayerProfile /> Should render properly 3`] = `
<div>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <img
      className="ui image"
      src=""
    />
    <div
      className="content"
    >
      <div
        className="header"
      >
        Marcos Alonso
        ,
        
         
        
      </div>
      <div>
        <div
          className="description"
        >
          <b>
            Pris: 
          </b>
          £
          -0.1
        </div>
        <div
          className="description"
        >
          <b>
            Poeng: 
          </b>
          -1
        </div>
        <div
          className="description"
        >
          <b>
            Valgt av: 
          </b>
          
          %
        </div>
        <div
          className="description"
        >
          <b>
            Skader: 
          </b>
          Ingen skader.
        </div>
        <div
          className="description"
        >
          <b>
            Sjanse for å spille neste runde:
          </b>
           
          -1
          %
        </div>
      </div>
    </div>
  </div>
</div>
`;
